### ISSUE TITLE
AttributeError: 'TOMLDocument' object has no attribute 'unwrap'

### ISSUE THREAD

**Issue by masbicudo** [pdm-project/pdm#1544]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

**Make sure you run commands with `-v` flag before pasting the output.**

Command that I ran:

```bash
pdm install -v
```

```
Traceback (most recent call last):
  File "C:\Users\masbi\.pyenv\pyenv-win\versions\3.10.6\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\masbi\.pyenv\pyenv-win\versions\3.10.6\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\users\masbi\appdata\roaming\python\scripts\pdm.exe\__main__.py", line 7, in <module>
  File "C:\Users\masbi\AppData\Roaming\pdm\venv\lib\site-packages\pdm\core.py", line 258, in main
    return Core().main(args)
  File "C:\Users\masbi\AppData\Roaming\pdm\venv\lib\site-packages\pdm\core.py", line 190, in main
    raise cast(Exception, err).with_traceback(traceback)
  File "C:\Users\masbi\AppData\Roaming\pdm\venv\lib\site-packages\pdm\core.py", line 185, in main
    f(project, options)
  File "C:\Users\masbi\AppData\Roaming\pdm\venv\lib\site-packages\pdm\cli\commands\install.py", line 62, in handle
    actions.do_sync(
  File "C:\Users\masbi\AppData\Roaming\pdm\venv\lib\site-packages\pdm\cli\actions.py", line 197, in do_sync
    candidates = resolve_candidates_from_lockfile(project, requirements)
  File "C:\Users\masbi\AppData\Roaming\pdm\venv\lib\site-packages\pdm\cli\actions.py", line 138, in resolve_candidates_from_lockfile
    provider = project.get_provider(for_install=True)
  File "C:\Users\masbi\AppData\Roaming\pdm\venv\lib\site-packages\pdm\project\core.py", line 427, in get_provider
    locked_repository = self.locked_repository
  File "C:\Users\masbi\AppData\Roaming\pdm\venv\lib\site-packages\pdm\project\core.py", line 392, in locked_repository
    lockfile = self.lockfile._data.unwrap()
AttributeError: 'TOMLDocument' object has no attribute 'unwrap'
```

## Steps to reproduce

**project.toml**
```
[project]
name = ""
version = ""
description = ""
authors = [
    {name = "Miguel Angelo", email = "masbicudo@gmail.com"},
]
dependencies = [
    "xgboost>=1.7.1",
    "pandas>=1.5.2",
]
requires-python = ">=3.8"
license = {text = "Unlicensed"}

[build-system]
requires = ["pdm-pep517>=1.0.0"]
build-backend = "pdm.pep517.api"

[tool]
[tool.pdm]
```

Run:

```bash
pdm install
```

## Actual behavior

Exception for which even Google has no clue about:

Search engine entry: "AttributeError" "TOMLDocument" "pdm"

## Expected behavior

Expected the installation of dependencies to complete successfully, or with an error message about no such combination of packages being possible... but not something that appears to be an internal library error.

## Environment Information

```bash
# Paste the output of `pdm info && pdm info --env` below:
PDM version:
  2.3.0
Python Interpreter:
  D:\Projects\Siemens\research\weaponization-2023\exploit-prediction-xgboost-n-classes\.venv\Scripts\python.exe (3.10)
Project Root:
  D:/Projects/Siemens/research/weaponization-2023/exploit-prediction-xgboost-n-classes
Project Packages:
  None
{
  "implementation_name": "cpython",
  "implementation_version": "3.10.6",
  "os_name": "nt",
  "platform_machine": "AMD64",
  "platform_release": "10",
  "platform_system": "Windows",
  "platform_version": "10.0.22621",
  "python_full_version": "3.10.6",
  "platform_python_implementation": "CPython",
  "python_version": "3.10",
  "sys_platform": "win32"
}
```

**Comment by frostming** [comment]
Fixed by #1539, update the version of tomlkit used by pdm

**Comment by masbicudo** [comment]
@frostming Thanks, I installed the HEAD version of PDM and the bug is indeed fixed.
